pilot/pkg/proxy/envoy/discovery.go

```diff
@@ -432,23 +442,40 @@ func (ds *DiscoveryService) clearCache() {
 	clearCacheMutex.Lock()
 	defer clearCacheMutex.Unlock()
 
-	if time.Since(lastClearCache) < time.Duration(clearCacheTime)*time.Second {
-		if !clearCacheTimerSet {
-			clearCacheTimerSet = true
-			time.AfterFunc(time.Duration(clearCacheTime)*time.Second, func() {
-				clearCacheMutex.Lock()
-				clearCacheTimerSet = false
-				clearCacheMutex.Unlock()
-				ds.clearCache() // it's after time - so will clear the cache
-			})
-		}
+	// If last config change was > 1 second ago, push.
+	if time.Since(lastClearCacheEvent) > 1*time.Second {
```

Comment:
How/why did we pick 1 second as the flush interval? Second, please make this a variable called something like `maxCacheClearInterval` and store it above with the cache vars.

Reply:
Good point - I'll do both on follow-up PRs (since the bot and CI are taking many hours to run, and we're trying to get this into the monthly ) - I don't think either change are critical (or worse than the bug )